### PR TITLE
Add more ltdetr configs

### PR DESCRIPTION
**Pull request by guarin**
## What has changed and why?

* Add DINOv3 B/L LTDETR configs
* Add DINOv2 B/L/G LTDETR configs

I updated the DINOv2 LTDETR ViT wrapper to match the DINOv3 one. All DINOv2 configs follow the settings from DINOv3 with the only difference in the patch size and num channels handling. Also deactivated STA for DINOv2 as it doesn't work properly.

This PR just makes the configs available. They are not yet optimized at all.

## How has it been tested?

* Manually

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [x] Yes
- [ ] Not needed (internal change)

## Did you update the documentation?

- [x] Yes
- [ ] Not needed (internal change without effects for user)

### CODE REVIEW

File: docs/source/object_detection.md [hunk 1]
```diff
@@ -30,7 +30,6 @@ using TensorRT version `10.13.3.9` and on a Nvidia T4 GPU with batch size 1.
 |  LightlyTrain  |     dinov3/vitt16-ltdetr-coco     |          49.8           |     5.4      |    10.1    |  640×640   |
 |  LightlyTrain  |   dinov3/vitt16plus-ltdetr-coco   |          52.5           |     7.0      |    18.1    |  640×640   |
 |  LightlyTrain  |     dinov3/vits16-ltdetr-coco     |          55.4           |     10.5     |    36.4    |  640×640   |
-|  LightlyTrain  |  dinov2/vits14-noreg-ltdetr-coco  |          55.7           |     16.9     |    55.3    |  644×644   |
```
Comment (guarin) [inline docs/source/object_detection.md:33]: Removed as this old checkpoint doesn't properly work anymore

File: src/lightly_train/_task_models/dinov2_ltdetr_object_detection/dinov2_vit_wrapper.py [hunk 2]
```diff
@@ -1,32 +1,241 @@
 #
-# Copyright (c) Lightly AG and affiliates.
-# All rights reserved.
```
Comment (guarin) [inline src/lightly_train/_task_models/dinov2_ltdetr_object_detection/dinov2_vit_wrapper.py:3]: This file is a copy-paste from the DINOv3 wrapper

File: src/lightly_train/_task_models/dinov2_ltdetr_object_detection/task_model.py [hunk 3]
```diff
@@ -40,6 +42,259 @@
 logger = logging.getLogger(__name__)
 
 
+class _HybridEncoderConfig(PydanticConfig):
+    in_channels: list[int]
+    feat_strides: list[int]
+    hidden_dim: int
+    use_encoder_idx: list[int]
+    num_encoder_layers: int
```
Comment (guarin) [inline src/lightly_train/_task_models/dinov2_ltdetr_object_detection/task_model.py:50]: Configs were copied from DINOv3

File: src/lightly_train/_task_models/dinov2_ltdetr_object_detection/train_model.py [hunk 4]
```diff
@@ -117,10 +120,9 @@ class DINOv2LTDETRObjectDetectionTrainArgs(TrainModelArgs):
     detector_weight_decay: float = 0.0
 
     # Scheduler configuration
-    scheduler_milestones: list[int] = Field(default_factory=lambda: [1000])
-    scheduler_gamma: float = 0.1
-    lr_warmup_steps: int | None = Field(
-        default=None,  # TODO (Thomas, 10/25): Change to flat-cosine with warmup.
+    scheduler_start_factor: float = 0.01
+    lr_warmup_steps: int = Field(
+        default=2000,
```
Comment (guarin) [inline src/lightly_train/_task_models/dinov2_ltdetr_object_detection/train_model.py:125]: Adjusted to use same optimizer, scheduler, and schedule as DINOv3

File: src/lightly_train/_task_models/dinov2_ltdetr_object_detection/train_model.py [hunk 5]
```diff
@@ -245,7 +248,6 @@ def training_step(
         self, fabric: Fabric, batch: ObjectDetectionBatch, step: int
     ) -> TaskStepResult:
         samples, boxes, classes = batch["image"], batch["bboxes"], batch["classes"]
-        boxes = _yolo_to_xyxy(boxes)
```
Comment (guarin) [inline src/lightly_train/_task_models/dinov2_ltdetr_object_detection/train_model.py:248]: This was a bug

File: src/lightly_train/_task_models/dinov3_ltdetr_object_detection/dinov3_vit_wrapper.py [hunk 6]
```diff
@@ -123,19 +123,18 @@ def __init__(
         model: DinoVisionTransformer,
         interaction_indexes: list[int] = [5, 8, 11],
         finetune: bool = True,
-        patch_size: int = 16,
```
Comment (guarin) [inline src/lightly_train/_task_models/dinov3_ltdetr_object_detection/dinov3_vit_wrapper.py:126]: Changed it to be inferred from the model instead

File: src/lightly_train/_task_models/dinov3_ltdetr_object_detection/task_model.py [hunk 7]
```diff
@@ -69,8 +68,8 @@ class _HybridEncoderConfig(PydanticConfig):
     upsample: bool = True
 
 
-class _HybridEncoderLargeConfig(_HybridEncoderConfig):
-    in_channels: list[int] = [384, 768, 1536]
+class _HybridEncoderTinyConfig(_HybridEncoderConfig):
+    in_channels: list[int] = [192, 384, 768]
     feat_strides: list[int] = [8, 16, 32]
```
Comment (guarin) [inline src/lightly_train/_task_models/dinov3_ltdetr_object_detection/task_model.py:73]: Didn't make any functional changes in the configs in this file. I only ordered the configs by model size from smallest to largest model

File: src/lightly_train/_task_models/dinov3_ltdetr_object_detection/task_model.py [hunk 8]
```diff
@@ -429,14 +520,20 @@ def __init__(
             ),
         }
         config_name = parsed_name["backbone_name"].replace("-notpretrained", "")
+        config_name = config_name.replace("-noreg", "")
         config_cls, wrapper_cls = config_mapping[config_name]
         config = config_cls()
 
         if hasattr(config, "backbone_wrapper"):
+            # TODO(Guarin, 02/26): Improve how mask tokens are handled for fine-tuning.
+            dinov3.mask_token.requires_grad = False  # type: ignore
```
Comment (guarin) [inline src/lightly_train/_task_models/dinov3_ltdetr_object_detection/task_model.py:529]: Small cleanup to align between DINOv2 and v3

File: src/lightly_train/_task_models/dinov3_ltdetr_object_detection/task_model.py [hunk 9]
```diff
@@ -460,13 +557,52 @@ def __init__(
 
     @classmethod
     def list_model_names(cls) -> list[str]:
-        # TODO: Lionel(09/25) Add support for ViT models as well.
         return [
-            f"{name}-{cls.model_suffix}"
-            for name in DINOV3_PACKAGE.list_model_names()
-            if "convnext" in name
+            f"{name}-{cls.model_suffix}" for name in DINOV3_PACKAGE.list_model_names()
```
Comment (guarin) [inline src/lightly_train/_task_models/dinov3_ltdetr_object_detection/task_model.py:561]: ViT models are now also supported

File: src/lightly_train/_task_models/dinov3_ltdetr_object_detection/task_model.py [hunk 10]
```diff
@@ -460,13 +557,52 @@ def __init__(
...
         ]
 
+    @classmethod
+    def is_supported_model(cls, model: str) -> bool:
+        try:
+            cls.parse_model_name(model_name=model)
+        except ValueError:
+            return False
+        else:
```
Comment (guarin) [inline src/lightly_train/_task_models/dinov3_ltdetr_object_detection/task_model.py:570]: Just changed order of methods. No functional changes.

File: src/lightly_train/_task_models/dinov3_ltdetr_object_detection/train_model.py [hunk 11]
```diff
@@ -158,7 +158,6 @@ def __init__(
         else:
             normalize_dict = normalize.model_dump()
 
-        self.model_state_dict_key_prefix = "model."
```
Comment (guarin) [inline src/lightly_train/_task_models/dinov3_ltdetr_object_detection/train_model.py:161]: Unused

File: src/lightly_train/_task_models/object_detection_components/hybrid_encoder.py [hunk 12]
```diff
@@ -390,7 +390,7 @@ def forward(self, feats):
             inner_outs[0] = feat_heigh
             if self.upsample:
                 upsample_feat = F.interpolate(
-                    feat_heigh, scale_factor=2.0, mode="nearest"
+                    feat_heigh, size=feat_low.shape[-2:], mode="nearest"
```
Comment (guarin) [inline src/lightly_train/_task_models/object_detection_components/hybrid_encoder.py:393]: More robust

File: src/lightly_train/_task_models/dinov2_ltdetr_object_detection/transforms.py [hunk 13]
```diff
@@ -68,19 +68,20 @@ class DINOv2LTDETRObjectDetectionRandomFlipArgs(RandomFlipArgs):
 
 
 class DINOv2LTDETRObjectDetectionScaleJitterArgs(ScaleJitterArgs):
+    # Sizes must be multiples of patch size * 2
     sizes: Sequence[tuple[int, int]] | None = [
-        (490, 490),
-        (518, 518),
-        (546, 546),
```
Comment (liopeer) [inline src/lightly_train/_task_models/dinov2_ltdetr_object_detection/transforms.py:74]: Don't quite get the motivation for changing this. These are already compatible with patch size 14.

Comment (guarin) [reply src/lightly_train/_task_models/dinov2_ltdetr_object_detection/transforms.py:74]: Images have to be divisible by `2*patch_size` for the STA. This is why I added the comment above. In the end I disable the STA but I still think it makes sense to keep the updated sizes.
